mods/creative/inventory.lua

```diff
@@ -191,10 +191,13 @@ function creative.register_tab(name, title, items)
 				inv.start_i = 0
 				inv.filter = ""
 				sfinv.set_player_inventory_formspec(player, context)
-			elseif fields.creative_search or
-					fields.key_enter_field == "creative_filter" then
+			elseif (fields.creative_search or
+					fields.key_enter_field == "creative_filter")
+					and fields.creative_filter then
 				inv.start_i = 0
-				inv.filter = fields.creative_filter:lower()
+				inv.filter = fields.creative_filter:sub(1, 1e3) -- truncate to a sane length
+						:gsub("[%z-\8\11-\31\127]", "") -- strip naughty control characters (keeps \t and \n)
```

Comment:
`\0` breaks Minetest in a few places (f.E. chat), leading to truncated strings (since there are plenty of places where Minetest calls `c_str()`). Not sure about the others, but I see no reason to allow them. Perhaps we should go even further and strip C1 control characters too (suggested by @luk3yx), but that seems nontrivial.

Reply:
Maybe the control character removal is unnecessary for the creative/craft guide search since the search query isn't logged anywhere or sent to other players?

Reply:
It is very likely overly cautious, but I don't see an issue with that. It just provides an additional thin ~~false sense of~~ security when somebody touches this code later on.